src/app/api/search.api.ts

```diff
@@ -106,38 +107,17 @@ export class SearchApi {
         if(id === '') {
             return of([]);
         }
-        const GET_DATA: DocumentNode = gql`
-{
-  search {
-    query(query: "${id}", perPage: 10) {
-      nodes {
-        ... on Dataset {
-          id
-        }
-      }
-    }
-  }
-}`
 
-        /* eslint-disable  @typescript-eslint/no-explicit-any */
-        // @ts-ignore
-      return this.apollo.watchQuery({query: GET_DATA})
-            .valueChanges.pipe(map((result: ApolloQueryResult<any>) => {
-                if (result.data) {
-                    return SearchApi.searchValueAddToAutocomplete(result.data.search.query.nodes || [], id);
-                } else {
-                    return [];
-                }
-            }));
-    }
-    private static searchValueAddToAutocomplete(ngTypeaheadList: DatasetIDsInterface[], searchValue: string): DatasetIDsInterface[] {
-        const newArray: DatasetIDsInterface[] = JSON.parse(JSON.stringify(ngTypeaheadList));
-        if (searchValue) {
-            newArray.unshift({__typename: TypeNames.allDataType, id: searchValue});
-        }
-        return newArray;
+        return this.searchAutocompleteGQL.watch({query: id, perPage: 10}).valueChanges.pipe(map(result => {
+            let results = result.data.search.query.nodes.map(
```

Comment:
Identifier `results` is never reassigned. Should be created new typization variable with let. For example:
```
let nodesList: DatasetIDsInterface [] = result.data.search.query.nodes.map(
v => ({id: v.name, __typename: v.__typename as TypeNames}));
if (searchValue) {
   nodesList.unshift({__typename: TypeNames.allDataType, id: id});
}
return nodesList;
```

Reply:
I think typechecking works just fine here. The function return type `Observable<DatasetIDsInterface[]>` already requires that `map()` operation has to produce `DatasetIDsInterface[]`.

As for:
```ts
if (searchValue) {
   nodesList.unshift({__typename: TypeNames.allDataType, id: id});
}
```
I intentionally changed the behavior - when you type `foobar` the search should still offer you an option to search for `foobar` in all datasets, even if autocomplete returned empty result.

Reply:
Ok. I agree